docs/howto/manage-web-app-charms/use-a-database.rst

```diff
@@ -0,0 +1,335 @@
+.. _use-db-with-12-factor-charms:
+
+Use a database with your 12-factor app charm
+============================================
+
+This guide demonstrates how to integrate a MySQL database into your FastAPI
+12-factor app. You will learn how to define object-relational mappings (ORM),
+handle schema migrations with Alembic, and manage the database integration using Juju.
+
+Prerequisites
+-------------
+
+- A working FastAPI application charmed with 12-factor tooling. This guide will be
+  following the :ref:`tutorial up until you deploy the app
+  <deploy-the-fastapi-app>`.
+- A Juju model
+- Basic familiarity with `SQLAlchemy <https://www.sqlalchemy.org/>`_ and
+  `Alembic <https://alembic.sqlalchemy.org/en/latest/>`_.
+
+
+Prepare the environment
+-----------------------
+
+MySQL requires a specific Python driver. Ensure your ``requirements.txt`` includes
+a MySQL driver compatible with SQLAlchemy, such as ``pymysql``.
```

Comment:
We give pymysql as an example, but we actually use it in the doc itself. Could we make it clear that we're using it for the rest of the how-to?